### DISCUSSION TITLE
Gatsby+Contentful+Netlify - occasionally missing images

### DISCUSSION THREAD

**Discussion by FrivalszkyP** [gatsbyjs/gatsby#38645] [Help]
Hi all,

I have the aforementioned configuration. When a new build is triggered by Contentful, sometimes, the Netlify build doesn't pick up the latest images added to my entries. Whether it be an inline asset in a rich text field or an image directly linked to an entry, it might end up missing from my build. When I clear the Netlify cache, the image appears. This is not Netlify-specific, though, I was able to produce the same issue on a local build. There could be something wrong with how the cache is handled by the contentful plugin. I'd love to start debugging it but I do not even know where to start. I'm using the latest gatsby ("5.12.7") and gatsby-source-contentful ("8.12.2") packages. 

Has anyone encountered anything similar?

**Comment by tomlschmidt** [comment]
I've been having issue like this with Asset references in general. And I've just had an occurrence, where a newly added image would only become visible after clearing the cache, then go missing after the next cached build, while usually it would stay visible once I had cleared the cache once.

I've found the `forceFullSync` option, but that has been removed in #33207, while still being mentioned in the README of gatsby-source-contentful, unfortunately.

Maybe @axe312ger can help us with this.

**Reply by FrivalszkyP** [reply]
I'm experimenting with the alpha version of the new contentful plugin but I have a long way to go before I can test the site with it, so I don't know if the plugin rewrite solves the issue and we should just wait for it. But I wouldn't be surprised if that was the official stance on this.

**Reply by axe312ger** [reply]
Hey you both.

Sorry for the late reply. The issues you have sound a lot like issues with syncing and Gatsby not picking up the changes correctly from Contentful.

Thanks for pointing out the forceFullSync, i'll remove it from the readme (It didnt work for quit a while anyways)

You already found out about the new version, it will significally improve hoe content is moved from Contentful into Gatsby. Please test the new version, if you do now, we can fix issues with YOUR project before we publish the new version. So upgrading will be easy for you then as well :) Feedback then here please: https://github.com/gatsbyjs/gatsby/discussions/38585

**Reply by tomlschmidt** [reply]
Thank you for your reply. I am in the process of testing the new version already and will try to keep you updated if the issues persist.
